internal/pkg/service/stream/storage/statistics/repository/stats_delete_test.go

```diff
@@ -312,4 +312,83 @@ func TestRepository_RollupStatisticsOnFileDelete_LevelTarget(t *testing.T) {
+	{
+		clk.Add(time.Hour)
+		require.NoError(t, statsRepo.ResetStats(sinkKey).Do(ctx).Err())
+		etcdhelper.AssertKVsFromFile(t, client, `fixtures/stats_delete_snapshot_004.txt`, etcdhelper.WithIgnoredKeyPattern(`^definition/|storage/file/|storage/slice/|storage/volume/`))
+	}
+
+	// Re-enable sink
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		// Enable sink, it opens a new file
+		require.NoError(t, defRepo.Sink().Enable(sinkKey, clk.Now(), by).Do(ctx).Err())
+	}
+
+	// Create another file/slice
+	// -----------------------------------------------------------------------------------------------------------------
+	var fileKey4 model.FileKey
+	var sliceKey4 model.SliceKey
+	{
+		files, err := fileRepo.ListIn(sinkKey).Do(ctx).All()
+		require.NoError(t, err)
+		require.Len(t, files, 1)
+		fileKey4 = files[0].FileKey
+
+		slices, err := sliceRepo.ListIn(sinkKey).Do(ctx).All()
+		require.NoError(t, err)
+		require.Len(t, slices, 1)
+		sliceKey4 = slices[0].SliceKey
+	}
+
+	// Create record
+	assert.NoError(t, statsRepo.Put(ctx, []statistics.PerSlice{
+		{
+			SliceKey: sliceKey4,
+			Value: statistics.Value{
+				SlicesCount:      1,
+				FirstRecordAt:    utctime.MustParse("2000-01-01T09:00:00.000Z"),
+				LastRecordAt:     utctime.MustParse("2000-01-01T10:00:00.000Z"),
+				RecordsCount:     1,
+				UncompressedSize: 1,
+				CompressedSize:   1,
+			},
+		},
+	}))
+
+	// Move statistics to the target level
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		// Disable sink, it triggers closing of the active file
+		require.NoError(t, defRepo.Sink().Disable(sinkKey, clk.Now(), by, "some reason").Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, sliceRepo.SwitchToUploading(sliceKey4, clk.Now()).Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, sliceRepo.SwitchToUploaded(sliceKey4, clk.Now()).Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.SwitchToImporting(fileKey4, clk.Now()).Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.SwitchToImported(fileKey4, clk.Now()).Do(ctx).Err())
+	}
+
+	// Check state
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		etcdhelper.AssertKVsFromFile(t, client, `fixtures/stats_delete_snapshot_005.txt`, etcdhelper.WithIgnoredKeyPattern(`^definition/|storage/file/|storage/slice/|storage/volume/`))
+	}
+
+	// Reset statistics
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		clk.Add(time.Hour)
+		require.NoError(t, statsRepo.ResetStats(sinkKey).Do(ctx).Err())
+		etcdhelper.AssertKVsFromFile(t, client, `fixtures/stats_delete_snapshot_006.txt`, etcdhelper.WithIgnoredKeyPattern(`^definition/|storage/file/|storage/slice/|storage/volume/`))
```

Comment:
I expected that also the final statistics is tested. I expect that all values are `0` after reset, but the test does not address that.